sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -84,6 +84,19 @@ typedef enum kmac_error {
   kKmacUnsupportedPaddingLength,
 } kmac_error_t;
 
+/**
+ * Return the matching enum of `kmac_key_len_t` for given key length.
+ *
+ * `key_len_enum` must not be NULL pointer.
+ *
+ * @param key_len The size of the key in bytes.
+ * @param key_len_enum The corresponding enum value to be returned.
+ * @return Error code.
+ */
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_get_key_len_bytes(size_t key_len,
+                                    kmac_key_len_t *key_len_enum);
```

Comment:
Isn't this function internal-only? Does it need to be exposed in the header file here?

Reply:
This is also called in `otcrypto_mac` in `mac.c`, in order to check whether there is an error with the key size passed with `key->config.key_length`. The returned enum result is discarded. I added the following comment lines:
```
  size_t key_len = key->config.key_length;
  ...
  // Call `kmac_get_key_len_bytes` to make sure that `key_len` is a valid key
  // size. Then discard the returned `key_len_enum` value.
  kmac_key_len_t key_len_enum;
  err = kmac_get_key_len_bytes(key_len, &key_len_enum);
  if (err != kKmacOk) {
    return kCryptoStatusBadArgs;
  }
```